### PR TITLE
signin-signup

### PR DESCRIPTION
**Pull request by AlselwiAisha**
- Add a login page where a User can create a user and login with with a username 
- Add authentication for the front-end to the server.

### CODE REVIEW

File: src/redux/user/userSlice.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+const initialState = {
+  auth_token: initialToken,
+  isLoggedIn: false,
+  user: {
+    id: null,
+    name: null,
+    email: null,
+    password: null,
+  },
```
Comment (Arch-Noize) [inline src/redux/user/userSlice.js:19]: Like I mentioned on another [PR](https://github.com/leloufadel/full-stuck-project/pull/26#pullrequestreview-1839581752), I am unsure why we are still using password if you've made changes on the backend to no longer use passwords.

File: src/components/user/login.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        user: {
+          email: loginEmail,
+          password: loginPassword,
+        },
+      };
+      try {
+        dispatch(loginUser(data));
+        if (isLoggedIn) {
+          navigate('/mainpage');
+          resetData();
```
Comment (Arch-Noize) [inline src/components/user/login.js]: There are a handful of weird errors when logging in. If one tries to enter with a user, logs out, and tries to log in again, the request fires off to log a user in regardless if they've finished writing their password. Here's the error I see on console after. 
image

Here are two instances of the error ocurring
(apologies for the audio on the video, forgot to turn off mic)

https://github.com/leloufadel/front-end-capstone/assets/91759364/7503b0d3-f16c-4ae2-9bb4-80e37dcd6517

https://github.com/leloufadel/front-end-capstone/assets/91759364/71a37421-d20a-4a0e-9313-540192f4aae4

Please look into this.

File: src/components/user/signup.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+              type="password"
+              className="input-text"
+              value={signUpPassword}
+              onChange={(e) => setSignUpPassword(e.target.value)}
+              placeholder="Password"
+              required
+            />
+            <br />
+            <input type="submit" value="Sign up" className="btn2" />
+            <input type="button" value="Cancel" className="btn2" onClick={handleCancelLogout} />
```
Comment (Arch-Noize) [inline src/components/user/signup.js:77]: Great work on adding the alerts to make the website notify you when a user has been successfully created! Same for the other notifications!!

**Review by Arch-Noize — CHANGES_REQUESTED** [review]
# Changes Required

Found an 'odd error' when logging in. It doesn't prevent a User from logging in... if anything it let's them login a little too fast. Please look at my comments below. They also concern logging in from the point of the backend

File: src/components/user/login.js (hunk 2 above)
Comment (AlselwiAisha) [reply src/components/user/login.js]: Ready.

**Review by Arch-Noize — APPROVED** [review]
# Approved! :green_circle: :green_circle: 

https://tenor.com/bbCiR.gif

Well done on implementing loging in. There's nothing left to do but to merge!

**Pull request merged**